### ISSUE TITLE
Discussion

### ISSUE THREAD

**Issue by ashmaroli** [kierenpitts/jekyll-issue-7811-example#1]
Based on the code in the plugin here, it looks like `NewsItemPage` and `NewsListingsPage` are just shells to render a pre-defined layout at a given URL.

In case you didn't already know, this is *almost* equivalent to having Markdown pages **with just front matter**:

```
---
layout: news_by_month
---

<end-of-file>
```

```
---
layout: news_browse
---

<end-of-file>
```

You could then use the plugin to `find` these `page` instances and inject `title` and other desired attributes to the instances..

**Comment by kierenpitts** [comment]
Thanks for the suggestion. I think this would work for the `news_browse` page but I'd need to generate each of the monthly listing pages on the fly so there will be an increasing number of these based on the plugin parsing the entries in the `_posts` directory and generating a monthly listing for each (the site I'm migrating to Jekyll and that I'm creating a plugin for has posts going back over 15 years and although there haven't been posts every month there have been posts published in more than 100 months). 

So although there's a 1:1 relationship between `news_browse` and `/news/browse.html` the layout for `news_by_month` will need to be used multiple times to generate a new page for each year and month combination - so a single Markdown page might not work for this unless I've misunderstood.

**Comment by ashmaroli** [comment]
In case you didn't know about it, there's already a plugin named `jekyll-archives`.. (You can either use it directly or go through its source-code for knowledge augmentation..)

Since the page contents are exactly the contents of your layout, I don't see need to *read* the layout multiple times. Reading it once and caching it in a class constant might be useful.
The basis of Liquid cache in Jekyll 4 is just one simple fact that the same Liquid template can be rendered multiple times with varying contexts. To illustrate:

```ruby
TEMPLATE = Liquid::Template.parse("hi {{name}}") # Parses, compiles and stashes the template

TEMPLATE.render('name' => 'John')                # => "hi John"
TEMPLATE.render('name' => 'Jane')                # => "hi Jane"
TEMPLATE.render('name' => 'John')                # => "hi John"
```
In other words, the 3 `render` calls above could be from 3 different pages without compiling the template multiple times.

**Comment by kierenpitts** [comment]
Thanks for the info. I think I'd misunderstood what you were suggesting in your first message so apologies for the confusion. Your comment makes sense and I think the reason my code was doing this was as I'd been creating the plugin based on the Generator example in the Jekyll docs. You're correct though, there's no need for me to compile the template multiple times and I'll change the plugin I'm creating to avoid the recompiling.

Thanks for mentioning the `jekyll-archives` plugin. I use that on another project and have been reading the source. It doesn't meet my needs exactly for the current project but has been a very useful reference.

Thanks again for the help and advice.

**Comment by ashmaroli** [comment]
The plugin in the docs is just an example. The best source of knowledge is always actual plugins in the wild. Either ways, good luck with your project.

P.S., you can delete this repo if you want.. (totally your call).

**Comment by kierenpitts** [comment]
I've updated the code and the README so it's clear what's going on and that it's unrelated to the other bug. It may be that the code/README/discussion will help someone else if they stumble across a similar issue and end up here.

Many thanks again.
